src/controllers/krate/publish.rs

```diff
@@ -49,14 +49,25 @@ pub fn publish(req: &mut dyn RequestExt) -> EndpointResult {
     let api_token_id = ids.api_token_id();
     let user = ids.user();
 
-    let verified_email_address = user.verified_email(&conn)?;
-    let verified_email_address = verified_email_address.ok_or_else(|| {
-        cargo_err(&format!(
-            "A verified email address is required to publish crates to crates.io. \
-             Visit https://{}/me to set and verify your email address.",
-            app.config.domain_name,
-        ))
-    })?;
+    let email_address = if app.config.require_email_verification {
+        let verified_email_address = user.verified_email(&conn)?;
+        verified_email_address.ok_or_else(|| {
+            cargo_err(&format!(
+                "A verified email address is required to publish crates to crates.io. \
+                 Visit https://{}/me to set and verify your email address.",
+                app.config.domain_name,
+            ))
+        })?
+    } else {
```

Comment:
Hm, I think what I would expect is that if `DISABLE_EMAIL_VERIFICATION_REQUIREMENT` is set, then we don't need to require the user to set *any* email.. that is, this `else` block shouldn't exist.